superset-frontend/packages/superset-ui-chart-controls/src/types.ts

```diff
@@ -237,7 +237,7 @@ export interface BaseControlConfig<
   ) => boolean;
   mapStateToProps?: (
     state: ControlPanelState,
-    controlState: ControlState,
+    controlState?: ControlState,
```

Comment:
@codyml why do you make this parameter optional?
This breaks all our custom override type checking.

Reply:
Hmm, I think that was a quick fix to stop the histogram viz plugin from complaining after I updated it to use shared controls.  I opened [a PR](https://github.com/apache/superset/pull/22014) that reverts that change and instead updates the plugin – does that look like a better solution?  Would be great to spin up an ephemeral env to check if there are any side-effects for Histogram charts.

Reply:
(Assuming it passes CI)

Reply:
https://github.com/apache/superset/pull/22014 was merged, let me know if that solves the issue!